File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestByte.js
print("getByte")
if(typeof getByte !== 'undefined')
	throw "getByte already defined"
getByte = callback("getByte")
getByteType = typeof getByte
print(getByteType)
print(getByte)
if(getByteType !== 'number')
	throw "getByte not number!"
callback("setByte",getByte)

print("getByteVector")
if(typeof getByteVector !== 'undefined')
	throw "getByteVector already defined"
getByteVector = callback("getByteVector")
getByteVectorType = typeof getByteVector[0]
print(getByteVectorType)
print(getByteVector)
if(getByteVectorType !== 'number')
	throw "getByteVector not number!"
callback("setByteVector",getByteVector)

print("getByteVectorAsList")
if(typeof getByteVectorAsList !== 'undefined')
	throw "getByteVectorAsList already defined"
getByteVectorAsList = callback("getByteVectorAsList")
getByteVectorAsListType = typeof getByteVectorAsList[0]
print(getByteVectorAsListType)
print(getByteVectorAsList)
if(getByteVectorAsListType !== 'number')
	throw "getByteVectorAsList not number!"
callback("setByteVectorAsList",getByteVectorAsList)

print("getByteMatrix")
if(typeof getByteMatrix !== 'undefined')
	throw "getByteMatrix already defined"
getByteMatrix = callback("getByteMatrix")
getByteMatrixType = typeof getByteMatrix[0][0]
print(getByteMatrixType)
print(getByteMatrix)
if(getByteMatrixType !== 'number')
	throw "getByteMatrix not number!"
callback("setByteMatrix",getByteMatrix)

print("getByteMatrixAsList")
if(typeof getByteMatrixAsList !== 'undefined')
	throw "getByteMatrixAsList already defined"
getByteMatrixAsList = callback("getByteMatrixAsList")
getByteMatrixAsListType = typeof getByteMatrixAsList[0][0]
print(getByteMatrixAsListType)
print(getByteMatrixAsList)
if(getByteMatrixAsListType !== 'number')
	throw "getByteMatrixAsList not number!"
callback("setByteMatrixAsList",getByteMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-ruby-contract/src/test/java/de/invesdwin/context/ruby/tests/callback/ParametersAndReturnsTestByte.rb
puts("getByte")
unless (defined?(getByte)).nil? || getByte.nil?
	raise "getByte already defined"
end
getByte = callback("getByte")
getByteType = getByte.class
puts(getByteType)
puts(getByte)
if(getByteType != Integer)
	raise "getByte not Integer!"
end
callback("setByte", getByte)

puts("getByteVector")
unless (defined?(getByteVector)).nil? || getByteVector.nil?
	raise "getByteVector already defined"
end
getByteVector = callback("getByteVector")
getByteVectorType = getByteVector[0].class
puts(getByteVectorType)
puts(getByteVector)
if(getByteVectorType != Integer)
	raise "getByteVector not Integer!"
end
callback("setByteVector", getByteVector)

puts("getByteVectorAsList")
unless (defined?(getByteVectorAsList)).nil? || getByteVectorAsList.nil?
	raise "getByteVectorAsList already defined"
end
getByteVectorAsList = callback("getByteVectorAsList")
getByteVectorAsListType = getByteVectorAsList[0].class
puts(getByteVectorAsListType)
puts(getByteVectorAsList)
if(getByteVectorAsListType != Integer)
	raise "getByteVectorAsList not Integer!"
end
callback("setByteVectorAsList", getByteVectorAsList)

puts("getByteMatrix")
unless (defined?(getByteMatrix)).nil? || getByteMatrix.nil?
	raise "getByteMatrix already defined"
end
getByteMatrix = callback("getByteMatrix")
getByteMatrixType = getByteMatrix[0][0].class
puts(getByteMatrixType)
puts(getByteMatrix)
if(getByteMatrixType != Integer)
	raise "getByteMatrix not Integer!"
end
callback("setByteMatrix", getByteMatrix)

puts("getByteMatrixAsList")
unless (defined?(getByteMatrixAsList)).nil? || getByteMatrixAsList.nil?
	raise "getByteMatrixAsList already defined"
end
getByteMatrixAsList = callback("getByteMatrixAsList")
getByteMatrixAsListType = getByteMatrixAsList[0][0].class
puts(getByteMatrixAsListType)
puts(getByteMatrixAsList)
if(getByteMatrixAsListType != Integer)
	raise "getByteMatrixAsList not Integer!"
end
callback("setByteMatrixAsList", getByteMatrixAsList)
